client/packages/programs/src/JsonForms/components/Search/Search.tsx

```diff
@@ -21,18 +22,18 @@ const UserOptions = z.object({
    * Pattern for formatting options list items (e.g. "${firstName} ${lastName}")
    */
   optionString: z.string().optional(),
-  /**
-   * Pattern for formatting selected result (as above)
-   */
-  displayString: z.string().optional(),
   /**
    * List of fields to save in document data (from selected item object)
    */
   saveFields: z.array(z.string()).optional(),
   /**
-   * Text to show in input field before user entry
+   * Child form elements
    */
-  placeholderText: z.string().optional(),
+  elements: z.array(z.any()),
+  /**
+   * List of fields to match against when searching Patients
+   */
+  searchFields: z.array(z.string()),
```

Comment:
Should we restrict which fields can be search for? I think you could do this with:
https://github.com/colinhacks/zod#sets
and with an zod enum for the `set`` type parameter:

`z.set(z.enum(["firstName", "lastName", ...]))`

Reply:
> Should we restrict which fields can be search for?

I was thinking that there would be no restriction by default. As per the [above comment](https://github.com/openmsupply/open-msupply/pull/2216#discussion_r1329467747), this component was intended to be able to other searches eventually, in which case the "Patient" fields are not the only possible search fields.

What do you think? Do we want to make this a generic search component (it can already do "by document" or "data lookup"), or will it only ever be for searching patients?

Reply:
Had another look, there is currently only the `"source": "patientContact"` option in use. This option displays an existing contact based on some preconfigure settings. On the other hand the patient search option searches a patient based on user input. I find it a bit confusing that one is preconfigured and the other uses' user input.

I am tending toward having a special `PatientSearch` component and another `DataLookup` or `DocumentData` component.

Reply:
> I am tending toward having a special `PatientSearch` component and another `DataLookup` or `DocumentData` component.

Yeah, but that's quite a big refactor though. Should it be a new issue, or change it here? Would want to do that first if we're going to go down that path.

Reply:
Yes agree can do this in another issue and have a chat about it if this is what we want...

Reply:
I added: https://github.com/openmsupply/open-msupply/issues/2279